iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/ProcessManager.java

```diff
@@ -239,16 +240,95 @@ public void deleteRemainingMicroservices() {
 		LoggingService.logInfo(MODULE_NAME ,"Finished delete Remaining Microservices");
 	}
 
-	private void deleteOldAgentContainers(Set<String> oldAgentContainerNames) {
-		logInfo("Delete old agent containers");
-		oldAgentContainerNames.forEach(name -> {
+	/**
+	 * Stop running microservices when agent deprovision.
+	 * Stop and delete microservices when agents stops
+	 * @param withCleanUp
+	 */
+	public void stopRunningMicroservices(boolean withCleanUp, String iofogUuid) {
+		LoggingService.logInfo(MODULE_NAME ,"Stop delete Remaining Microservices");
+		if (withCleanUp) {
+			List<Container> allContainers;
+			synchronized (deleteLock) {
+				allContainers = docker.getContainers();
+			}
+			Set<String> allMicroserviceUuids = allContainers
```

Comment:
Shouldn't we check labels here as well?

Reply:
In this flow the agent is deprovisioned already and the value of `Configuration.getIofogUuid` is set to blank. Therefore I have skipped the checking of label.